core/trino-main/src/main/java/io/trino/operator/join/LookupJoinOperatorFactory.java

```diff
@@ -99,12 +99,20 @@ public LookupJoinOperatorFactory(
             this.outerOperatorFactory = Optional.empty();
         }
         else {
-            this.outerOperatorFactory = Optional.of(new LookupOuterOperatorFactory(
-                    operatorId,
-                    planNodeId,
-                    probeOutputTypes,
-                    buildOutputTypes,
-                    lookupSourceFactoryManager));
+            List<Type> types = ImmutableList.<Type>builder()
+                    .addAll(probeOutputTypes)
+                    .addAll(buildOutputTypes)
+                    .build();
```

Comment:
We're duplicating logic from `LookupOuterOperatorFactory`.
(did you mean to use `LookupOuterOperatorFactory.getTypes()` also added in this commit?)

Maybe, instead, we could have an optional method on OperatorFactory 
```
Optional<List<Type>> returnedTypes();
```

then, also maybe, we wouldn't need the `TypedOperatorFactory`.